src/main/java/pathfinding/ConcreteNode.java

```diff
@@ -77,6 +77,13 @@ public void addNeighbor(Node newNeighbor)
 		neighbors.add(newNeighbor);
 	}
 
+	public void removeNeighbor(Node oldNeighbor)
+	{
+		if (neighbors.contains(oldNeighbor)) {
```

Comment:
Brace placement. Not a problem for this pull request, I'll just go through the code on friday (or thursday afternoon when there's no work being done) and redo some of the formatting.